### PR TITLE
Improve FT.INFO complexity to O(1)

### LINKED ISSUE #2116: Remove `TrieMap_MemUsage` from `FT.INFO`
Instead, maintain a counter for memory usage.
This will make `FT.INFO` time complexity `O(1)`.

### PR DESCRIPTION
**Pull request by ashtul**
Fixes #2116 

The previous function iterated recursively over the whole trie and collected the size of each node including the string which defines the node ID. This operation has the complexity of O(n) and presents the lower bound of memory allocation since the memory allocated for the struct would be padded to 8 bytes (since the struct contains `void *` ptr).

The calculation in this PR is based on the cardinality of the trie. It multiplies it by the size of the struct, the point and char held by its parent, and a constant of 8 bytes, to compensate for the memory which is padded out.

### CODE REVIEW

**Review by rafie — DISMISSED** [review]

File: tests/pytests/test_issues.py [hunk 1]
```diff
@@ -236,3 +236,33 @@ def test_MOD1266(env):
   env.expect('FT.SEARCH', 'jsonidx', 'redis').equal([1L, '1', ['$', '{"t":"Redis"}']])
   conn.execute_command('JSON.SET', '1', '$.t', r'{"inner_t":"Redis"}')
   env.expect('FT.SEARCH', 'jsonidx', '*').equal([0L])
+
+def testMemAllocated(env):
+  conn = getConnectionByEnv(env)
```
Comment (MeirShpilraien) [inline tests/pytests/test_issues.py:241]: I think it's a good idea to verify that the test passes on the coordinator and if not try to adjust it.

File: src/dep/triemap/triemap.h [hunk 2]
```diff
@@ -45,7 +45,8 @@ typedef struct {
 
 typedef struct {
   TrieMapNode *root;
-  size_t cardinality;
+  size_t cardinality : 32;  // number of terms
```
Comment (MeirShpilraien) [inline src/dep/triemap/triemap.h:48]: Why limit it to 32 bit? it's not like we have so many tries that it will consume too much memory no?

**Review by MeirShpilraien — COMMENTED** [review]
Small comments, looks very good :+1:

File: tests/pytests/test_issues.py (hunk 1 above)
Comment (ashtul) [reply tests/pytests/test_issues.py:241]: Checked and passed

**Review by MeirShpilraien — APPROVED** [review]
:+1:

**Pull request merged**